cmd/globaldns.go

```diff
@@ -0,0 +1,865 @@
+		if err != nil {
+			return "", err
+		}
+		target = fmt.Sprintf("Multi Cluster App %s", app.Name)
+	} else {
+		var targets []managementClient.Target
+		for _, projectID := range entry.ProjectIDs {
+			targets = append(targets, managementClient.Target{
+				ProjectID: projectID,
+			})
+		}
+		targetNames := getReadableTargetNames(clusterCache, projectCache, targets)
+		target = strings.Join(targetNames, ",")
+	}
+	return target, nil
+}
+
+func listGlobalDNSMembers(ctx *cli.Context) error {
+	if ctx.NArg() == 0 {
+		return cli.ShowSubcommandHelp(ctx)
+	}
+
+	c, err := GetClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	entry, err := searchForGlobalDNS(c, ctx.Args().First())
+	if nil != err {
+		return err
+	}
+
+	return outputMembers(ctx, c, entry.Members)
+}
+
+func globalDNSCreate(ctx *cli.Context) error {
+	c, err := GetClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	fqdn := ctx.String(argFQDN)
+	provider := ctx.String(argProvider)
+	appName := ctx.String(argMultiClusterApp)
+	projects := ctx.StringSlice(argProject)
+	ttl := ctx.Int64(argTTL)
+
+	if fqdn == "" {
+		return errors.New("--fqdn is required")
+	}
+
+	if provider == "" {
+		return errors.New("--provider is required")
+	}
+
+	if appName != "" && len(projects) > 0 || appName == "" && len(projects) == 0 {
+		return fmt.Errorf("please specify either --multi-cluster-app or --project as the global DNS entry target")
+	}
+
+	entry := &managementClient.GlobalDNS{
+		FQDN: fqdn,
+		TTL:  ttl,
+	}
+
+	globalDNSProvider, err := searchForGlobalDNSProvider(c, provider)
+	if err != nil {
+		return err
+	}
+	entry.ProviderID = globalDNSProvider.ID
+
+	members, err := addMembersByNames(ctx, c, entry.Members, ctx.StringSlice(argMember), memberAccessTypeOwner)
+	if err != nil {
+		return err
+	}
+	entry.Members = members
+
+	if appName != "" {
+		resource, err := Lookup(c, appName, managementClient.MultiClusterAppType)
+		if err != nil {
+			return err
+		}
+		entry.MultiClusterAppID = resource.ID
+	} else {
+		projectIDs, err := lookupProjectIDsFromTargets(c, projects)
+		if err != nil {
+			return err
+		}
+		entry.ProjectIDs = projectIDs
+	}
+
+	if _, err = c.ManagementClient.GlobalDNS.Create(entry); nil != err {
+		return err
+	}
+
+	fmt.Printf("Successfully created global DNS entry for %q\n", fqdn)
+	return nil
+}
+
+func globalDNSUpdate(ctx *cli.Context) error {
+	if ctx.NArg() == 0 {
+		return cli.ShowSubcommandHelp(ctx)
+	}
+
+	c, err := GetClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	entry, err := searchForGlobalDNS(c, ctx.Args().First())
+	if err != nil {
+		return err
+	}
+
+	update := make(map[string]interface{})
+
+	if ctx.IsSet(argFQDN) {
+		update["fqdn"] = ctx.String(argFQDN)
+	}
+
+	if ctx.IsSet(argProvider) {
+		globalDNSProvider, err := searchForGlobalDNSProvider(c, ctx.String(argProvider))
+		if err != nil {
+			return err
+		}
+		update["providerId"] = globalDNSProvider.ID
+	}
+
+	if ctx.IsSet(argMultiClusterApp) {
```

Comment:
I think you need to validate that we can not support both `argProvider` and `argMultiClusterApp `  target in one update command.

Reply:
Why can't we do that?

Reply:
nvm, I see it now,